lib/enigma.rb

```diff
@@ -1,16 +1,43 @@
+require "date"
 require "./lib/message_processor"
 class Enigma
 
   def output_hash(processed_message, key, date, action = :encryption)
     Hash[action, processed_message, :key, key, :date, date]
   end
 
-  def encrypt(input_message, key, date)
+  def random_number
+    (0..9).to_a.sample(5).shuffle.join
+  end
+
+  def todays_date
+    Date.today.strftime("%d%m%y")
+  end
+
+  def valid_date?(date)
+    return false unless date.length == 6
+    return false unless date.split("").all?{ |num| ("0".."9").include?(num)}
+    Date.valid_date?("20#{date[-2,2]}".to_i,date[2,2].to_i, date[0,2].to_i)
+  end
+
+  def valid_key?(key)
+    return false unless key.length == 5
+    return false unless key.split("").all?{ |num| ("0".."9").include?(num)}
+    true
```

Comment:
i can remove the 'true' on line 26, remove 'return false unless' on line 25, the #all? returns a boolean